ReactNativeClient/lib/models/Folder.js

```diff
@@ -119,12 +122,23 @@ class Folder extends BaseItem {
 			${where} GROUP BY folders.id`;
 
 		const noteCounts = await this.db().selectAll(sql);
+
+		// this function is to get the note_count
+		noteCounts.forEach((noteCount) => {
+			const parentId = noteCount.folder_id;
+			const folder = foldersById[parentId];
+			if (folder) {
+				folder.note_count = (folder.note_count || 0) + noteCount.note_count;
+			}
+		});
+
+		// this function is to get the note_count_all (count of notes in this and subfolders)
 		noteCounts.forEach((noteCount) => {
 			let parentId = noteCount.folder_id;
 			do {
 				const folder = foldersById[parentId];
 				if (!folder) break; // https://github.com/laurent22/joplin/issues/2079
-				folder.note_count = (folder.note_count || 0) + noteCount.note_count;
+				folder.note_count_all = (folder.note_count_all || 0) + noteCount.note_count;
```

Comment:
This value can be derived from note_count from the view, so it shouldn't be attached to the model. You can create a helper function to calculate this value if needed, but just don't save it to the model.

Reply:
And please add tests for that helper function.

Reply:
> This value can be derived from note_count from the view, so it shouldn't be attached to the model. You can create a helper function to calculate this value if needed, but just don't save it to the model.

Sir, I looked at what you were saying, but sir I have some doubts. Sir, if I make a separate function to get note_count_all for each folder in view, it will call that helper function every time component gets re-render,  that is basically every time someone clicks on any folder or open and close it.
And one thing sir, In each folder, I don't have reference to its children's, I only have reference to the only parent of every folder. So if I want to calculate note_count_all of a folder, I can't just go its children and add the note_count, instead, I have to call a recursive function, which will check for every possible folder for each layer of subfolders or the one way can be that I made an array of all temp folders and then attach note_count_all to them just like I did now and then return only the folder which I required. I don't think both methods will be very efficient for so many folders.
So can you suggest me a better way of doing this?

Reply:
The performance issue is not a problem as that can be optimised later on if needed.

> And one thing sir, In each folder, I don't have reference to its children's, I only have reference to the only parent of every folder.

Yes but wherever the note_count property is used, you also have the complete array of folders nearby, which you can use to dynamically calculate the note_count_all value.